packages/vue-renderer/src/renderer.js

```diff
@@ -1,26 +1,34 @@
 import path from 'path'
 import crypto from 'crypto'
-import devalue from '@nuxtjs/devalue'
-import template from 'lodash/template'
 import fs from 'fs-extra'
-import { createBundleRenderer } from 'vue-server-renderer'
 import consola from 'consola'
-
+import devalue from '@nuxtjs/devalue'
+import template from 'lodash/template'
 import { waitFor } from '@nuxt/common'
+import { matchesUA } from 'browserslist-useragent'
+import { createBundleRenderer } from 'vue-server-renderer'
 
+import ModernBrowsers from '../data/modern-browsers.json'
 import SPAMetaRenderer from './spa-meta'
 
 export default class VueRenderer {
   constructor(context) {
     this.context = context
 
     // Will be set by createRenderer
-    this.bundleRenderer = null
-    this.spaMetaRenderer = null
+    this.renderer = {
+      ssr: null,
+      modern: null,
+      spa: null
+    }
+
+    this.modernBrowsers = Object.keys(ModernBrowsers)
```

Comment:
Can't we precompute this and put what we cant into JSON file? (To reduce startup time)